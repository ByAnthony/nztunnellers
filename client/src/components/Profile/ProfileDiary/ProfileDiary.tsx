import {
  MilitaryYears,
  Origins, PostServiceYears, PreWayYears,
} from '../../../types/tunneller';

import { DiaryArmyExperience } from './DiaryArmyExperience/DiaryArmyExperience';
import { DiaryArrivedInNz } from './DiaryArrivedInNz/DiaryArrivedInNz';
import { DiaryBirth } from './DiaryBirthInfo/DiaryBirthInfo';
import { DiaryDied } from './DiaryDied/DiaryDied';
import { DiaryHometown } from './DiaryHometown/DiaryHometown';
import { DiaryLife } from './DiaryLife/DiaryLife';
import { DiaryMedal } from './DiaryMedal/DiaryMedal';
import { DiaryParents } from './DiaryParents/DiaryParents';
import { DiaryWork } from './DiaryWork/DiaryWork';

import STYLES from './ProfileDiary.module.scss';

type Props = {
  tunnellerId: number;
  origins: Origins;
  preWarYears: PreWayYears;
  militaryYears: MilitaryYears;
  postWarYears: PostServiceYears;
}

export function ProfileDiary({
  tunnellerId,
  origins,
  preWarYears,
  militaryYears,
  postWarYears,
}: Props) {
  return (
    <div className={STYLES.diary}>
      <h2>About</h2>
      <DiaryBirth birth={origins.birth} />
      <DiaryParents parents={origins.parents} />
      <DiaryArrivedInNz inNzLength={origins.inNzLength} />
      <DiaryHometown residence={preWarYears.residence} />
      <DiaryWork employment={preWarYears.employment} />
      <DiaryLife maritalStatus={preWarYears.maritalStatus} wife={preWarYears.wife} />
      <DiaryArmyExperience tunnellerId={tunnellerId} armyExperience={preWarYears.armyExperience} />
      <DiaryMedal medals={militaryYears.medals} />
      <DiaryDied
        warDeath={militaryYears.endOfService.deathWar}
        afterWarDeath={postWarYears.death}
      />
    </div>
  );
}
