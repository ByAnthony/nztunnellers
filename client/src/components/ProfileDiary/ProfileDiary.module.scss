@use 'src/index.scss';

.diary {
  width: 100%;
  margin-bottom: index.$m-margin;
  font-weight: 600;
  color: index.$text-color;
  text-align: left;

  p {
    margin: 0;
  }

  .fullwidth-main-card {
    @include index.fullwidth-main-card;
  }

  .fullwidth-secondary-card {
    @include index.fullwidth-secondary-card;
  }

  .halfwidth-cards-container {
    @include index.halfwidth-cards-container;
  }

  .halfwidth-main-card {
    @include index.halfwidth-card-style;
    @include index.halfwidth-main-card;
  }

  .halfwidth-secondary-card {
    @include index.halfwidth-card-style;
    @include index.halfwidth-secondary-card;
  }

  .halfwidth-secondary-card-title {
    @include index.halfwidth-secondary-card-title;
  }
}
