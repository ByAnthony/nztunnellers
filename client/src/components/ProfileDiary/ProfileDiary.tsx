import {
  Date, Origins, Parent, Parents,
} from '../../types/tunneller';
import STYLES from './ProfileDiary.module.scss';

type Props = {
  origins: Origins;
}

export function ProfileDiary({ origins }: Props) {
  const displayBirthCountry = (country: string | null) => (
    country !== null ? (
      <p>
        {`Born in ${country}`}
      </p>
    ) : <p>Born</p>
  );

  const displayBirthDate = (birth: Date) => (
    birth.dayMonth !== null ? (
      `${birth.dayMonth} ${birth.year}`
    ) : `${birth.year}`
  );

  const displayParents = (parents: Parents) => {
    if (parents.mother !== null && parents.father !== null) {
      return (
        <>
          <div className={STYLES['fullwidth-title-card']}>
            Parents
          </div>
          <div className={STYLES['card-container']}>
            <div className={STYLES['parents-card']}>
              <div className={STYLES['parents-card-title']}><p>Mother</p></div>
              <div>{ parents.mother.name }</div>
            </div>
            <div className={STYLES['parents-card']}>
              <div className={STYLES['parents-card-title']}><p>Father</p></div>
              <div>{ parents.father.name }</div>
            </div>
          </div>
        </>
      );
    }

    const isMotherOrFather = parents.mother ? 'Mother' : 'Father';

    const displayParent = (parent: Parent) => (
      <div className={STYLES['card-container']}>
        <div className={STYLES['parent-title-card']}>
          Parent
        </div>
        <div className={STYLES['parent-card']}>
          <div className={STYLES['parents-card-title']}><p>{ isMotherOrFather }</p></div>
          <div>{ parent.name }</div>
        </div>
      </div>
    );

    if (parents.mother !== null && parents.father === null) {
      return displayParent(parents.mother);
    }
    if (parents.mother === null && parents.father !== null) {
      return displayParent(parents.father);
    }
    return null;
  };

  return (
    <div className={STYLES.diary}>
      <h2>Diary</h2>
      <div className={STYLES['fullwidth-title-card']}>
        { displayBirthCountry(origins.birth.country) }
        { displayBirthDate(origins.birth.date) }
      </div>
      { displayParents(origins.parents) }
    </div>
  );
}
