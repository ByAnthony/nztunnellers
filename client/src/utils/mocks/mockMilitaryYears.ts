import {
  Date,
  DeathPlace,
  Death,
  DeathCause,
  Cemetery,
  Medal,
  MilitaryYears,
  Enlistment,
  EmbarkationUnit,
  Training,
  Transport,
  EndOfService,
  Demobilization,
} from '../../types/tunneller';
import { mockFrontEvents } from './mockFrontEvents';

export const mockBritishWarMedal: Medal = {
  name: 'British War Medal',
  country: 'United Kingdom',
  image: 'british-war-medal.png',
  citation: null,
};

export const mockVictoryMedal: Medal = {
  name: 'Victory Medal',
  country: 'United Kingdom',
  image: 'victory-medal.png',
  citation: null,
};

export const mockMedals: Medal[] = [
  mockBritishWarMedal,
  mockVictoryMedal,
];

export const mockDate: Date = {
  year: '1926',
  dayMonth: '13 October',
};

const mockPlace: DeathPlace = {
  location: 'Telegraph Hill',
  town: 'Arras',
  country: 'France',
};

const mockCause: DeathCause = {
  type: 'Killed in action',
  circumstances: 'Dead wounded by shell fire',
};

const mockCemetery: Cemetery = {
  name: 'Faubourg d\'Amiens',
  location: 'Arras',
  country: 'France',
  graveReference: 'IE 18',
};

export const mockDeath: Death = {
  date: mockDate,
  place: mockPlace,
  cause: mockCause,
  cemetery: mockCemetery,
  ageAtDeath: 89,
};

export const mockEnlistment: Enlistment = {
  serial: '1/1000',
  rank: 'Sapper',
  date: mockDate,
  district: 'Canterbury',
  alias: null,
  transferredToTunnellers: null,
};

const mockTraining: Training = {
  date: mockDate,
  location: 'Auckland',
  locationType: 'Military Camp',
};

export const mockEmbarkationUnit: EmbarkationUnit = {
  detachment: 'Main Body',
  training: mockTraining,
  section: 'Section No.2',
  attachedCorps: null,
};

export const mockTransport: Transport = {
  reference: 'HMNZT 56',
  vessel: 'Beata',
  departureDate: mockDate,
  departurePort: 'Auckland',
  arrivalDate: mockDate,
  arrivalPort: 'Plymouth',
};

const mockDemobilization: Demobilization = {
  date: mockDate,
  country: 'New Zealand',
};

const mockEndOfService: EndOfService = {
  deserter: false,
  transferred: null,
  deathWar: null,
  transportNz: mockTransport,
  demobilization: mockDemobilization,
};

export const mockMilitaryYears: MilitaryYears = {
  enlistment: mockEnlistment,
  embarkationUnit: mockEmbarkationUnit,
  transportUk: mockTransport,
  frontEvents: mockFrontEvents,
  endOfService: mockEndOfService,
  medals: mockMedals,
};
