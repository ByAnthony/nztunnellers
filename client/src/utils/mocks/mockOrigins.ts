import { Birth, Origins, Parents } from '../../types/tunneller';

export const mockBirth: Birth = {
  date: {
    year: '1886',
    dayMonth: '18 December',
  },
  country: 'New Zealand',
};

export const mockParents: Parents = {
  mother: {
    name: 'Jane Doe',
    origin: 'New Zealand',
  },
  father: {
    name: 'John Doe',
    origin: 'Scotland',
  },
};

export const mockOrigins: Origins = {
  birth: mockBirth,
  parents: mockParents,
  inNzLength: null,
};
