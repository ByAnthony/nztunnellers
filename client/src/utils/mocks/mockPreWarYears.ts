import { ArmyExperience, Employment, PreWayYears } from '../../types/tunneller';

export const mockArmyExperience = (overrides: Partial<ArmyExperience>
    | undefined = undefined): ArmyExperience => ({
  unit: 'NZ Infantry',
  country: 'New Zealand',
  conflict: null,
  duration: '12 months',
  ...overrides,
});

export const mockEmployment = (overrides: Partial<Employment>
    | undefined = undefined): Employment => ({
  occupation: 'Goldminer',
  employer: 'Goldmining Company',
  ...overrides,
});

export const mockPreWarYears = (overrides: Partial<PreWayYears>
    | undefined = undefined): PreWayYears => ({
  armyExperience: [mockArmyExperience()],
  employment: mockEmployment(),
  residence: null,
  maritalStatus: null,
  wife: null,
  religion: null,
  ...overrides,
});
