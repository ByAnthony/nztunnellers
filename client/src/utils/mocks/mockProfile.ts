// import { Profile } from '../../types/tunneller';
// import { mockSummary } from './mockSummary';

export const mockId = 26;

// export const mockProfile = (overrides: Partial<Profile> | undefined = undefined): Profile => ({
//   id: mockId,
//   summary: mockSummary(),
//   origins: mockOrigins(),
//   preWarYears: mockPreWarYears(),
//   militaryYears: mockMilitaryYears(),
//   postServiceYears: mockPostServiceYears(),
//   sources: mockSources(),
//   image: mockImage(),
//   ...overrides,
// });
