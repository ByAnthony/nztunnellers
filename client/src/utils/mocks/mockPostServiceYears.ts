import { PostServiceYears, PostWarDeath } from '../../types/tunneller';
import { mockDeath } from './mockMilitaryYears';

export const mockAfterWarDeath: PostWarDeath = {
  deathWarInjury: false,
  ...mockDeath,
};

export const mockPostServiceYears: PostServiceYears = {
  death: mockAfterWarDeath,
};
