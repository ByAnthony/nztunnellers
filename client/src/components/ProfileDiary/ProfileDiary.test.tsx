import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { mockId } from '../../utils/mocks/mockProfile';
import { mockMilitaryYears } from '../../utils/mocks/mockMilitaryYears';
import { mockOrigins } from '../../utils/mocks/mockOrigins';
import { mockPostServiceYears } from '../../utils/mocks/mockPostServiceYears';
import { mockPreWarYears } from '../../utils/mocks/mockPreWarYears';
import { ProfileDiary } from './ProfileDiary';

const component = (
  <ProfileDiary
    tunnellerId={mockId}
    origins={mockOrigins}
    preWarYears={mockPreWarYears}
    militaryYears={mockMilitaryYears}
    postWarYears={mockPostServiceYears}
  />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});
