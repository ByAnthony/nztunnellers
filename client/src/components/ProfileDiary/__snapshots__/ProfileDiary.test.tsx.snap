// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="diary"
  >
    <h2>
      About
    </h2>
    <div
      class="fullwidth-main-card"
    >
      <p>
        Born in New Zealand
      </p>
      <span>
        18 December 1886
      </span>
    </div>
    <div
      class="fullwidth-main-card"
    >
      Parents
    </div>
    <div
      class="halfwidth-cards-container"
    >
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            Mother
          </p>
        </div>
        <div>
          <span>
            Jane Doe
          </span>
        </div>
      </div>
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            Father
          </p>
        </div>
        <div>
          <span>
            John Doe
          </span>
        </div>
      </div>
    </div>
    <div
      class="fullwidth-main-card"
    >
      Work
    </div>
    <div
      class="halfwidth-cards-container"
    >
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            Occupation
          </p>
        </div>
        <div>
          <span>
            Goldminer
          </span>
        </div>
      </div>
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            Employer
          </p>
        </div>
        <div>
          <span>
            Goldmining Company
          </span>
        </div>
      </div>
    </div>
    <h2>
      Army Experience
    </h2>
    <li
      class="fullwidth-secondary-card"
    >
      <span>
        NZ Infantry
      </span>
      <p>
        12 months in New Zealand
      </p>
    </li>
    <a
      aria-label="Open the World War I timeline"
      class="war-service"
      href="/tunnellers/26/wwi-timeline"
    >
      <div>
        <p>
          World War I (1914-1918)
        </p>
        <span>
          New Zealand Tunnellers
        </span>
      </div>
      <div
        class="arrow-right"
      >
        â†’
      </div>
    </a>
    <div
      class="halfwidth-cards-container"
    >
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            <img
              alt="British War Medal ribbon"
              src="/images/roll/medals/british-war-medal.png"
              width="40"
            />
          </p>
        </div>
        <div>
          <span>
            British War Medal
          </span>
        </div>
      </div>
      <div
        class="halfwidth-secondary-card"
      >
        <div
          class="halfwidth-secondary-card-title"
        >
          <p>
            <img
              alt="Victory Medal ribbon"
              src="/images/roll/medals/victory-medal.png"
              width="40"
            />
          </p>
        </div>
        <div>
          <span>
            Victory Medal
          </span>
        </div>
      </div>
    </div>
    <h2>
      Death
    </h2>
    <div
      class="fullwidth-main-card"
    >
      <p>
        Died at the age of 89
      </p>
      <span>
        13 October 1926
      </span>
    </div>
  </div>
</DocumentFragment>
`;
